newrelic/structures_newrelic_synthetics_simple_browser_monitor.go

```diff
@@ -69,22 +72,26 @@ func buildSyntheticsSimpleBrowserMonitor(d *schema.ResourceData) synthetics.Synt
 	do, doOk := d.GetOk("device_orientation")
 	dt, dtOk := d.GetOk("device_type")
 
-	if doOk || dtOk {
-		simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation = &synthetics.SyntheticsDeviceEmulationInput{}
-
-		if doOk {
-			simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation.DeviceOrientation = synthetics.SyntheticsDeviceOrientation(do.(string))
-		}
+	if !(scriptLangOk && runtimeTypeOk && runtimeTypeVersionOk) && (doOk && dtOk) {
+		return simpleBrowserMonitorInput, errors.New("device emulation is not supported by legacy runtime")
+	}
 
-		if dtOk {
-			simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation.DeviceType = synthetics.SyntheticsDeviceType(dt.(string))
+	if doOk && dtOk {
+		simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation = &synthetics.SyntheticsDeviceEmulationInput{}
+		simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation.DeviceOrientation = synthetics.SyntheticsDeviceOrientation(do.(string))
+		simpleBrowserMonitorInput.AdvancedOptions.DeviceEmulation.DeviceType = synthetics.SyntheticsDeviceType(dt.(string))
+	} else {
+		if doOk || dtOk {
+			return simpleBrowserMonitorInput, errors.New("both device_orientation and device_type should be specified to enable device emulation")
 		}
 	}
 
-	return simpleBrowserMonitorInput
+	return simpleBrowserMonitorInput, nil
 }
 
-func buildSyntheticsSimpleBrowserMonitorUpdateStruct(d *schema.ResourceData) synthetics.SyntheticsUpdateSimpleBrowserMonitorInput {
+// TODO: Reduce the cyclomatic complexity of this func
+// nolint: gocyclo
```

Comment:
I think splitting this function into two can help reduce cyclomatic complexity of this function, for instance, we can shift all of the logic that deals with runtime values and device emulation into a new function. Can you give this a try and see if it works? If not, we can then consider this temporary workaround until we find a better solution